### PR TITLE
Remove beta guard from telemetry SPI

### PR DESCRIPTION
**Pull request by benjamin-confino**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Associated with epic: https://github.ibm.com/websphere/WS-CD-Open/issues/32050

### CODE REVIEW

File: dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/interfaces/OpenTelemetryAccessor.java [hunk 1]
```diff
@@ -42,11 +42,10 @@ public class OpenTelemetryAccessor {
      *         is disabled or the application has shut down.
      */
     public static OpenTelemetryInfoInternal getOpenTelemetryInfo() {
-        //TOOD when the SPI is out of beta, make this a redirect to the SPI accessor to avoid code duplication
```
Comment (benjamin-confino) [inline dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/interfaces/OpenTelemetryAccessor.java:45]: I did this inverted, as most calls will come through this class so redirecting into it saves one more hop for the most used path.

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Review by Azquelt — CHANGES_REQUESTED** [review]
Changes in this PR look ok, but we can't consider merging it at the moment because the related tests are only testing against `mpTelemetry-2.0` and not `1.0` or `1.1`.

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
The force push was removing the commit to update the repeat tests, that was delivered under https://github.com/OpenLiberty/open-liberty/pull/30071

**Comment by benjamin-confino** [comment]
#build

**Comment by contbld** [comment]
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=40d149bb-45e6-472c-a242-8933b7a74c9e

Target locations of links might be accessible only to IBM employees.

**Review by Azquelt — APPROVED** [review]

**Comment by benjamin-confino** [comment]
The lite fats have finished with no failures in the telemetry fats.

**Comment by contbld** [comment]
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=40d149bb-45e6-472c-a242-8933b7a74c9e has errors/failures.

**Pull request merged**